test/Sema/enum_raw_representable_generic.swift

```diff
@@ -0,0 +1,23 @@
+// RUN: %target-run-simple-swift
+
+// REQUIRES: executable-test
+
+enum Foo<T : ExpressibleByIntegerLiteral> : T where T:Equatable {
+  case a = 1
+  case b = 10
+  case c = 42
+}
+
+let _ : Int8  = Foo.a.rawValue
+let _ : Int16 = Foo.a.rawValue
+let _ : Int32 = Foo.a.rawValue
+let _ : Int64 = Foo.a.rawValue
+let _ : UInt8  = Foo.a.rawValue
+let _ : UInt16 = Foo.a.rawValue
+let _ : UInt32 = Foo.a.rawValue
+let _ : UInt64 = Foo.a.rawValue
+
+guard case .c = Foo(rawValue: UInt8(42))!  else { fatalError() }
+guard case .c = Foo(rawValue: UInt16(42))! else { fatalError() }
+guard case .c = Foo(rawValue: UInt32(42))! else { fatalError() }
+guard case .c = Foo(rawValue: UInt64(42))! else { fatalError() }
```

Comment:
Thanks -- what happens currently if the types don't match up? You should add a test case either way.

No worries, the codebase is huge so we're all learning :)